superset-frontend/src/features/alerts/AlertReportModal.tsx

```diff
@@ -776,6 +804,25 @@ const AlertReportModal: FunctionComponent<AlertReportModalProps> = ({
     [],
   );
 
+  const dashboard = currentAlert?.dashboard;
+  useEffect(() => {
+    if (!tabsEnabled) return;
+
+    if (dashboard?.value) {
+      SupersetClient.get({
+        endpoint: `/api/v1/dashboard/${dashboard.value}/tabs`,
+      }).then(response => {
+        const { tab_tree: tabTree, all_tabs: allTabs } = response.json.result;
+        setTabOptions(tabTree);
+        if (currentAlert?.extra?.dashboard?.anchor) {
+          if (!(currentAlert?.extra?.dashboard?.anchor in allTabs)) {
```

Comment:
You are already checking above if this exists. Maybe it would help readability to move it inside a const.

```
const anchor = currentAlert?.extra?.dashboard?.anchor
if (anchor && !anchor in allTabs)...
```